lib/data_migrate/database_tasks.rb

```diff
@@ -74,14 +74,22 @@ def schema_dump_path(db_config, format = ActiveRecord.schema_format)
         # We only require a schema.rb file for the primary database
         return unless db_config.primary?
 
-        File.join(File.dirname(ActiveRecord::Tasks::DatabaseTasks.schema_dump_path(db_config, format)), schema_file_type)
+        if Gem::Dependency.new("railties", ">= 6.1").match?("railties", Gem.loaded_specs["railties"].version)
+          File.join(File.dirname(ActiveRecord::Tasks::DatabaseTasks.schema_dump_path(db_config, format)), schema_file_type)
+        else
+          super.gsub(/(_)?schema\.rb\z/, '\1data_schema.rb')
+        end
       end
 
       # Override this method from `ActiveRecord::Tasks::DatabaseTasks`
       # to ensure that the sha saved in ar_internal_metadata table
       # is from the original schema.rb file
       def schema_sha1(file)
-        ActiveRecord::Tasks::DatabaseTasks.schema_dump_path(ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env, name: "primary"))
+        if Gem::Dependency.new("railties", ">= 6.1").match?("railties", Gem.loaded_specs["railties"].version)
+          ActiveRecord::Tasks::DatabaseTasks.schema_dump_path(ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env, name: "primary"))
+        else
+          super(file.gsub(/data_schema.rb\z/, 'schema.rb'))
+        end
```

Comment:
I noticed that the version check for railties is repeated in both methods, to improve readability and reduce duplication, how about refactoring it like this:

```rb
def schema_dump_path(db_config, format = ActiveRecord.schema_format)
  return ENV["DATA_SCHEMA"] if ENV["DATA_SCHEMA"]
  return unless db_config.primary?  # We only require a schema.rb file for the primary database

  if railties_version < Gem::Version.new('6.1')
    return super.gsub(/(_)?schema\.rb\z/, '\1data_schema.rb')
  end

  File.join(
    File.dirname(ActiveRecord::Tasks::DatabaseTasks.schema_dump_path(db_config, format)),
    schema_file_type
  )
end

def schema_sha1(file)
  if railties_version < Gem::Version.new('6.1')
    return super(file.gsub(/data_schema.rb\z/, 'schema.rb'))
  end

  ActiveRecord::Tasks::DatabaseTasks.schema_dump_path(
    ActiveRecord::Base.configurations.configs_for(
      env_name: ActiveRecord::Tasks::DatabaseTasks.env,
      name: "primary"
    )
  )
end

private

def railties_version
  Gem.loaded_specs["railties"].version
end
```

Reply:
Drying and early returns are nice, I'll implement that. I'll leave the existing sort of syntax for checking the gem versions as that's how we've checked in other places